components/tidb_query/src/rpn_expr/impl_time.rs

```diff
@@ -209,4 +226,40 @@ mod tests {
             .unwrap();
         assert_eq!(output, None);
     }
+
+    #[test]
+    fn test_to_days() {
+        let cases = vec![
+            ("950501", Some(728779)),
+            ("2007-10-07", Some(733321)),
+            ("2008-10-07", Some(733687)),
+            ("08-10-07", Some(733687)),
+            ("0000-01-01", Some(1)),
+            ("2007-10-07 00:00:59", Some(733321)),
+        ];
+        let mut ctx = EvalContext::default();
+        for (arg, exp) in cases {
+            let time = Some(Time::parse_datetime(&mut ctx, arg, 6, true).unwrap());
+            let output = RpnFnScalarEvaluator::new()
+                .push_param(time.clone())
+                .evaluate(ScalarFuncSig::ToDays)
+                .unwrap();
+            assert_eq!(output, exp);
+        }
+
+        // test NULL case
+        let output = RpnFnScalarEvaluator::new()
+            .push_param(None::<Time>)
+            .evaluate::<Int>(ScalarFuncSig::ToDays)
+            .unwrap();
+        assert_eq!(output, None);
+
+        // test ZERO case
+        let time = Some(Time::parse_datetime(&mut ctx, "0000-00-00 00:00:00", 6, true).unwrap());
```

Comment:
Any case to test invalid datetime value?

Reply:
Hi,  `Time::parse_datetime(...)` will get panic when test invalid datetime, so maybe we can't input invalid datetime?